src/Toolkit/Toolkit.Maui/BookmarksView/BookmarksView.cs

```diff
@@ -192,16 +170,36 @@ private void SelectAndNavigateToBookmark(Bookmark bookmark)
     /// <summary>
     /// Handles selection on the underlying list view.
     /// </summary>
-    private void Internal_bookmarkSelected(object? sender, SelectedItemChangedEventArgs e)
+    private void Internal_bookmarkSelected(object? sender, SelectionChangedEventArgs e)
+    {
+        if (sender is CollectionView cv)
+        {
+            if (cv.SelectedItem is Bookmark item)
+            {
+                SelectAndNavigateToBookmark(item);
+            }
+            cv.ClearValue(CollectionView.SelectedItemProperty);
+        }
+    }
+
+    private void UpdateListView()
     {
-        if (e.SelectedItem is Bookmark bm)
+        var collection = (CollectionView)GetTemplateChild(_presentingViewName);
+        if (collection != null)
         {
-            SelectAndNavigateToBookmark(bm);
+            collection.SelectionChanged -= Internal_bookmarkSelected;
+            collection.SelectionChanged += Internal_bookmarkSelected;
+            collection.ItemTemplate = null;
+            collection.ItemTemplate = ItemTemplate;
+            collection.ItemsSource = _dataSource;
         }
+    }
 
-        if (e.SelectedItem != null && sender is ListView lv)
+    private void OnBookmarksCollectionChanged(object? sender, NotifyCollectionChangedEventArgs e)
+    {
+        if (e.Action is NotifyCollectionChangedAction.Add)
         {
-            lv.SelectedItem = null;
+            UpdateListView();
         }
     }
```

Comment:
This should be in `#if Android` since that's the only thing using it. However I think cleaning up the event handling will aviod this workaround